client/src/controllers/UpgradeController.test.js

```diff
@@ -0,0 +1,133 @@
+
+    fetch.mockResponseSuccess(JSON.stringify(data));
+
+    const application = Application.start();
+
+    application.register('w-upgrade', UpgradeController);
+
+    // trigger next browser render cycle
+    await Promise.resolve(true);
+
+    expect(global.fetch).toHaveBeenCalledWith(
+      'https://releases.wagtail.org/mock.txt',
+      { referrerPolicy: 'strict-origin-when-cross-origin' },
+    );
+    expect(
+      document.getElementById('panel').classList.contains('w-hidden'),
+    ).toBe(true);
+
+    await flushPromises();
+
+    // should remove the hidden class on success
+    expect(
+      document.getElementById('panel').classList.contains('w-hidden'),
+    ).toBe(false);
+
+    // should update the latest version number in the text
+    expect(document.getElementById('latest-version').innerText).toBe(
+      data.version,
+    );
+
+    // should add the latest version link
+    expect(document.getElementById('link').getAttribute('href')).toEqual(
+      data.minorUrl,
+    );
+  });
+
+  it.skip('should not show the message if the current version is up to date', async () => {
+    const data = {
+      version: '5.15.1',
+      url: 'https://docs.wagtail.org/latest/url',
+      minorUrl: 'https://docs.wagtail.org/latest-minor/url',
+      lts: {
+        version: '2.2',
+        url: 'https://docs.wagtail.org/lts/url',
+        minorUrl: 'https://docs.wagtail.org/lts-minor/url',
+      },
+    };
+
+    fetch.mockResponseSuccess(JSON.stringify(data));
+
+    expect(global.fetch).not.toHaveBeenCalled();
+
+    const application = Application.start();
+
+    // trigger next browser render cycle
+    await Promise.resolve(true);
+
+    application.register('w-upgrade', UpgradeController);
+
+    expect(
+      document.getElementById('panel').classList.contains('w-hidden'),
+    ).toBe(true);
+  });
+
+  it('should throw an error if the fetch fails', () => {
+    const application = Application.start();
+
+    application.register('w-upgrade', UpgradeController);
+
+    // Spy on console.error to verify that it is called with the expected error message
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+
+    // Create a mock for the fetch function that returns a rejected Promise
+    const mockFetch = jest.fn(() => Promise.reject(new Error('Fetch failed')));
+    // Replace the global fetch function with the mock
+    global.fetch = mockFetch;
+
+    // Wait for the catch block to be executed
+    /* eslint-disable-next-line no-promise-executor-return */
+    return new Promise((resolve) => setImmediate(resolve)).then(() => {
+      // Verify that console.error was called with the expected error message
+      /* eslint-disable-next-line no-console */
+      expect(console.error).toHaveBeenCalledWith(
+        `Error fetching ${url}. Error: Error: Fetch failed`,
+      );
+
+      // Restore the original implementation of console.error
+      /* eslint-disable-next-line no-console */
```

Comment:
You can probably just disable the Eslint no console directive for this whole file.